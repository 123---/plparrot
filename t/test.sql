\set ECHO
\set QUIET 1
-- Turn off echo and keep things quiet.
-- Format the output for nice TAP.
\pset format unaligned
\pset tuples_only true
\pset pager
-- Revert all changes on failure.
\set ON_ERROR_ROLLBACK 1
\set ON_ERROR_STOP true
\set QUIET 1
-- Load the TAP functions.
BEGIN;
    \i pgtap.sql
-- Plan the tests.
SELECT plan(2);

--DROP FUNCTION plparrot_call_handler() CASCADE;
-- TODO: Make this configurable
INSERT INTO pg_catalog.pg_pltemplate(
    tmplname,
    tmpltrusted,
    tmpldbacreate,
    tmplhandler,
    tmpllibrary
)
VALUES (
    'plparrot',
    true,
    true,
    'plparrot_call_handler',
    '$libdir/plparrot'
);
CREATE LANGUAGE plparrot;

-- These functions should be written in PIR

CREATE FUNCTION test_void() RETURNS void AS $$ FAIL $$ LANGUAGE plparrot;

CREATE FUNCTION test_int() RETURNS int AS $$ select 1 as result $$ LANGUAGE plparrot;

CREATE FUNCTION test_int_int(int) RETURNS int AS $$ select $1 as result $$ LANGUAGE plparrot;

CREATE FUNCTION test_float() RETURNS float AS $$ select 1.0 as result $$ LANGUAGE plparrot;

CREATE FUNCTION test_varchar() RETURNS varchar(5) AS $$ select 'cheese' as result $$ LANGUAGE plparrot;

select test_void();
select is(test_int(),1);
select is(test_int_int(42),42);

-- There does not seem to be any floating point comparison functions in pgTAP
--select like(test_float(), 1.0,1e-6);
select is(test_varchar(), 'cheese');

-- Finish the tests and clean up.
SELECT * FROM finish();

rollback;
