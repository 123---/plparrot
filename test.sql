DROP FUNCTION plparrot_call_handler() CASCADE;
CREATE FUNCTION plparrot_call_handler() RETURNS LANGUAGE_HANDLER AS 'plparrot' LANGUAGE C;
CREATE LANGUAGE plparrot HANDLER plparrot_call_handler;
