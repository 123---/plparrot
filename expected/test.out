\unset ECHO
1..31
ok 1 - we can .include PIR libraries included with Parrot
ok 2 - we can load_bytecode PBC libraries included with Parrot
ok 3 - FileHandle.open is mocked in PL/PIR
ok 4 - FileHandle.open is not mocked in PL/PIRU
ok 5 - File.open is mocked
ok 6 - We can pass a text in
ok 7 - We can return a text
ok 8 - We can pass a varchar in
ok 9 - We can return a varchar
ok 10 - We can concat and return a varchar
ok 11 - We can pass an int and float as arguments
ok 12 - We can pass a char in
ok 13 - We can return a char
ok 14 - We can pass in an int
ok 15 - We can return an int
ok 16 - plpir is an alias for plparrot
ok 17 - plpiru can return values
ok 18 - plparrotu can return values
ok 19 - Immutable works
ok 20 - Strict works
ok 21 - We can increment an int and return it
ok 22 - We can return an int
ok 23 - We can return void
ok 24 - We can return a float
ok 25 - We can add to a float and return it
ok 26 - We can pass a timestamp in
ok 27 - We can return a timestamp
ok 28 - We can pass a timestamptz in
ok 29 - We can return a timestamptz
ok 30 - We can pass a time in
ok 31 - We can return a time
